test/e2e_env/kubernetes/meshtimeout/meshtimeout.go

```diff
@@ -17,94 +20,66 @@ import (
-		)).To(Succeed())
-	})
-
-	AfterEachFailure(func() {
-		DebugKube(kubernetes.Cluster, mesh, namespace)
-	})
-
-	E2EAfterAll(func() {
-		Expect(kubernetes.Cluster.TriggerDeleteNamespace(namespace)).To(Succeed())
-		Expect(kubernetes.Cluster.DeleteMesh(mesh)).To(Succeed())
-	})
-
-	DescribeTable("should add timeouts", FlakeAttempts(3), func(timeoutConfig string) {
-		// given no MeshTimeout
-		mts, err := kubernetes.Cluster.GetKumactlOptions().KumactlList("meshtimeouts", mesh)
-		Expect(err).ToNot(HaveOccurred())
-		Expect(mts).To(
-			Or(
-				HaveExactElements(Equal(fmt.Sprintf("mesh-gateways-timeout-all-%s.kuma-system", mesh))),
-				BeEmpty(),
-			),
-		)
-
-		Eventually(func(g Gomega) {
-			start := time.Now()
-			_, err := client.CollectEchoResponse(
-				kubernetes.Cluster, "demo-client", testServerURL,
-				client.FromKubernetesPod(namespace, "demo-client"),
-				client.WithHeader("x-set-response-delay-ms", "5000"),
-				client.WithMaxTime(10),
-			)
-			g.Expect(err).ToNot(HaveOccurred())
-			g.Expect(time.Since(start)).To(BeNumerically(">", time.Second*5))
-		}, "30s", "1s").Should(Succeed())
-
-		// when
-		Expect(YamlK8s(timeoutConfig)(kubernetes.Cluster)).To(Succeed())
-
-		// then
-		Eventually(func(g Gomega) {
-			response, err := client.CollectFailure(
-				kubernetes.Cluster, "demo-client", testServerURL,
-				client.FromKubernetesPod(namespace, "demo-client"),
-				client.WithHeader("x-set-response-delay-ms", "5000"),
-				client.WithMaxTime(10), // we don't want 'curl' to return early
-			)
-			g.Expect(err).ToNot(HaveOccurred())
-			g.Expect(response.ResponseCode).To(Equal(504))
-		}, "1m", "1s", MustPassRepeatedly(5)).Should(Succeed())
-
-		Expect(DeleteYamlK8s(timeoutConfig)(kubernetes.Cluster)).To(Succeed())
-	},
-		Entry("outbound timeout", fmt.Sprintf(`
+	DescribeTableSubtree("with meshServices mode", func(mode mesh_proto.Mesh_MeshServices_Enabled) {
+		mesh := fmt.Sprintf("meshtimeout-ms-%s", strings.ToLower(mode.String()))
+		namespace := fmt.Sprintf("%s-namespace", mesh)
+		testServerURL := fmt.Sprintf("test-server.%s.svc:80", namespace)
+
+		BeforeAll(func() {
+			err := NewClusterSetup().
+				Install(YamlK8s(builders.Mesh().
+					WithName(mesh).
+					WithoutInitialPolicies().
+					WithMeshServicesEnabled(mode).
+					KubeYaml())).
+				Install(NamespaceWithSidecarInjection(namespace)).
+				Install(democlient.Install(democlient.WithNamespace(namespace), democlient.WithMesh(mesh))).
+				Install(testserver.Install(testserver.WithMesh(mesh), testserver.WithNamespace(namespace))).
+				Setup(kubernetes.Cluster)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEachFailure(func() {
+			DebugKube(kubernetes.Cluster, mesh, namespace)
+		})
+
+		E2EAfterAll(func() {
+			Expect(kubernetes.Cluster.TriggerDeleteNamespace(namespace)).To(Succeed())
+			Expect(kubernetes.Cluster.DeleteMesh(mesh)).To(Succeed())
+		})
+
+		DescribeTable("should add timeouts", FlakeAttempts(3), func(timeoutConfig string) {
```

Comment:
Why do we need `FlakeAttempts(3)` here? Is that still flaky?

Reply:
It was there so left it